mlrun/api/db/sqldb/db.py

```diff
@@ -240,6 +240,20 @@ def _store_artifact(
             tag = tag or "latest"
             self.tag_artifacts(session, [art], project, tag)
 
+    def _add_tag_to_artifact_struct(
+        self, session, artifact_struct, artifact_id, tag=None
+    ):
+        if tag:
+            artifact_struct["tag"] = tag
+        else:
+            tag_object = self._query(
+                session, Artifact.Tag, obj_id=artifact_id
+            ).one_or_none()
```

Comment:
If the artifact will have several tag you will explode here - handle that + add test for such scenario

Reply:
Fixed.